.github/workflows/release-azure-pipelines.yml

```diff
@@ -93,40 +93,268 @@ extends:
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              rootFolderOrFile: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}
+              includeRootFolder: false
+              archiveType: zip
+              archiveFile: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.zip
+          - task: EsrpCodeSigning@5
+            displayName: Sign artifacts osx
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              ConnectedServiceName: $(ESRP_KV_SERVICE_CONNECTION)
+              AppRegistrationClientId: $(SIGNING_AAD_ID)
+              AppRegistrationTenantId: $(SIGNING_TENANT_ID)
+              AuthAKVName: $(AZURE_VAULT)
+              AuthCertName: $(AZURE_VAULT_ESRP_AAD_CERT_NAME)
+              AuthSignCertName: $(AZURE_VAULT_ESRP_REQ_CERT_NAME)
+              FolderPath: $(Build.ArtifactStagingDirectory)
+              Pattern: 'azureauth-${{ parameters.version }}-${{ config.runtime }}.zip'
+              signConfigType: 'inlineSignParams'
+              inlineOperation: |
+                [
+                  {
+                    "KeyCode": "$(SIGNING_KEY_CODE_MAC)",
+                    "OperationCode": "MacAppDeveloperSign",
+                    "ToolName": "sign",
+                    "ToolVersion": "1.0",
+                    "Parameters": {}
+                 },
+                 {
+                  "KeyCode": "$(SIGNING_KEY_CODE_MAC)",
+                  "OperationCode": "SigntoolVerify",
+                  "ToolName": "sign",
+                  "ToolVersion": "1.0",
+                  "Parameters": {}
+                 }
+                ]
+              SessionTimeout: '60'
+              MaxConcurrency: '50'
+              MaxRetryAttempts: '5'
+              PendingAnalysisWaitTimeoutMinutes: '5'
+          - task: ExtractFiles@1
+            displayName: Extract signed artifacts osx
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              archiveFilePatterns: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.zip
+              destinationFolder: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}
+              cleanDestinationFolder: true
+              overwriteExistingFiles: true
+          # We rename the signed artifacts to avoid conflicts with the unsigned pipeline artifacts from the previous stage.
+          - task: PowerShell@2
+            displayName: Rename signed artifacts
+            inputs:
+              workingDirectory: $(Build.ArtifactStagingDirectory)
+              targetType: 'inline'
+              script: |
+                mv "azureauth-${{ parameters.version }}-${{ config.runtime }}" "azureauth-${{ parameters.version }}-${{ config.runtime }}-signed"

```

Comment:
Do you run into conflicts if you use the same file name, but a different artifact name?

Reply:
Do you mean different artificat name in the `output`? If so, I tried doing that, but it couldn't find the path. It searches for the exact file.

Reply:
Yep I meant exactly that! And that's surprising! I always thought of artifacts as different "buckets" of files. I wonder if maybe there's a "bug" in how 1ES PT consume artifacts via the "inputs" template context. But either way, not a big deal :)